lib/puppet/provider/parsedfile.rb

```diff
@@ -449,7 +440,7 @@ def flush
     # If the target isn't set, then this is our first modification, so
     # mark it for flushing.
     unless @property_hash[:target]
-      @property_hash[:target] = @resource.should(:target) || self.class.default_target
```

Comment:
I think this may be because I started to use `target` as a parameter instead of a property.